txscript/sign.go

```diff
@@ -383,7 +383,10 @@ func mergeScripts(chainParams *chaincfg.Params, tx *wire.MsgTx, idx int,
 		// We already know this information somewhere up the stack.
 		class, addresses, nrequired, err :=
 			ExtractPkScriptAddrs(DefaultScriptVersion, script, chainParams)
-
+		if err != nil {
```

Comment:
This does not look accurate to me.  This implies that if I provide a script that can't be signed, it's going to panic.  That does not at all seem like desired behavior.

Reply:
Based on the comment it appears that this function has already run higher in the call stack with the same inputs, so erroring here now where it didn't error before is unexpected.

Reply:
mergeScripts doesn't return error

Reply:
that shouldn't be a reason to keep the panic here, but it is definitely why it was ignored before

Reply:
I agree with jrick.  I would prefer the error be ignored and reorganized in a separate PR.  I don't like the introduction of the panic here.

Reply:
no problem.  I switched to _ and made the comment even more explicit.